bundles/org.openhab.binding.doorbird/src/main/resources/ESH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,224 @@
+		<label>Image Timestamp</label>
+		<description>Time when image was captured from device</description>
+		<state readOnly="true"></state>
+	</channel-type>
+	<channel-type id="image">
+		<item-type>Image</item-type>
+		<label>Image</label>
+		<description>Image from device</description>
+		<state readOnly="true"></state>
+	</channel-type>
+	<channel-type id="doorbellHistoryIndex">
+		<item-type>Number</item-type>
+		<label>Doorbell History Index</label>
+		<description>Index of historical image for doorbell press</description>
+		<state min="1" max="50" step="1"></state>
+	</channel-type>
+	<channel-type id="doorbellHistoryTimestamp">
+		<item-type>DateTime</item-type>
+		<label>Doorbell History Timestamp</label>
+		<description>Time when doorbell was pressed for history image</description>
+		<state readOnly="true"></state>
+	</channel-type>
+	<channel-type id="doorbellHistoryImage">
+		<item-type>Image</item-type>
+		<label>Doorbell History Image</label>
+		<description>Historical image for doorbell press</description>
+		<state readOnly="true"></state>
+	</channel-type>
+	<channel-type id="motionHistoryIndex">
+		<item-type>Number</item-type>
+		<label>Motion History Index</label>
+		<description>Index of Historical image for motion</description>
+		<state min="1" max="50" step="1"></state>
+	</channel-type>
+	<channel-type id="motionHistoryTimestamp">
+		<item-type>DateTime</item-type>
+		<label>Motion History Timestamp</label>
+		<description>Time when motion was detected for history image</description>
+		<state readOnly="true"></state>
+	</channel-type>
+	<channel-type id="motionHistoryImage">
+		<item-type>Image</item-type>
+		<label>Motion History Image</label>
+		<description>Historical image for motion sensor</description>
+		<state readOnly="true"></state>
+	</channel-type>
+	<channel-type id="getDoorbellMontage" advanced="false">
+		<item-type>Switch</item-type>
+		<label>Get Doorbell Montage</label>
+		<description>Refresh the doorbell montage</description>
+		<state pattern="%s"></state>
+	</channel-type>
+	<channel-type id="doorbellMontage">
+		<item-type>Image</item-type>
+		<label>Doorbell Montage Image</label>
+		<description>Montage of multiple doorbell history images</description>
+		<state readOnly="true"></state>
+	</channel-type>
+	<channel-type id="getMotionMontage" advanced="false">
+		<item-type>Switch</item-type>
+		<label>Get Motion Montage</label>
+		<description>Refresh the motion montage</description>
+		<state pattern="%s"></state>
+	</channel-type>
+	<channel-type id="motionMontage">
+		<item-type>Image</item-type>
+		<label>Motion Montage Image</label>
+		<description>Montage of multiple motion history images</description>
+		<state readOnly="true"></state>
+	</channel-type>
+	<channel-type id="sipHangup">
+		<item-type>Switch</item-type>
+		<label>SIP Hangup</label>
+		<description>Hangup a SIP call</description>
+		<state pattern="%s"></state>
+	</channel-type>
+	<channel-type id="restart">
```

Comment:
Can we model this - the restart channel - as a thing action?

Reply:
Yes, it certainly could be modeled that way. I looked at the VolvoOnCall binding, and it looks pretty straightforward to set it up as an action. I'm only wondering if it makes sense to do that if `restart` is the only action supported by the binding. WDYT?

Reply:
Yes, why not?  It can be a simple action without any parameters.

And maybe in the future there will be more :wink:. Point is I would like to push some of the new features in OH and thing actions can be used with the new rule engine (NGRE).

Reply:
Ok, that seems reasonable. I'll change this to a thing action.

Reply:
> And maybe in the future there will be more

I converted the `restart` and `sipHangup` channels to thing actions. I put the action package as a subpackage of `doorbird` (versus `doorbird.internal`). I saw bindings do it both ways, so I'm not sure which is preferred.

Reply:
Looks good. :+1: I'd tend to internal.

Reply:
Hmm. When I move the `action` package under `internal`, I'm not able to invoke the action from within a rule. the rule throws an error that says
```
'sipHangup' is not a member of 'org.eclipse.smarthome.core.thing.binding.ThingActions'
```

Is there something else I need to do to cause the actions to be available in a rule? Does the package need to be exported somehow?

Reply:
Since packages that match `*.internal.*` are not exported, I don't think this will work.

Reply:
I was wondering about that. But there are bindings that have their `action` package under `internal`. See below. Then those actions should not work either?

https://github.com/openhab/openhab2-addons/blob/9c2629cd0bdd53276a9e3439bb52cd83f278627c/bundles/org.openhab.binding.volvooncall/src/main/java/org/openhab/binding/volvooncall/internal/action/VolvoOnCallActions.java

https://github.com/openhab/openhab2-addons/blob/86191c59572ee16852edc77170de0829f675a0b0/bundles/org.openhab.binding.pushbullet/src/main/java/org/openhab/binding/pushbullet/internal/action/PushbulletActions.java

https://github.com/openhab/openhab2-addons/blob/b3125023fe01fd1474e25e1eb68227e9d40ad289/bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/action/LightActions.java

Reply:
IIRC there is a spacial handling inside the core framework to expose `ThingActions` towards DSL (see https://github.com/openhab/openhab-core/blob/20faf1c63b02f150501f81995db39e7b0d5d1bb3/bundles/org.openhab.core.model.script/src/org/eclipse/smarthome/model/script/scoping/ScriptImplicitlyImportedTypes.java#L129-L134). Please correct me if I am wrong.

The problem we might face here is that the import will be done once during start if the DSL rule engine. Meaning if you add a binding later during runtime an new action might not be available immediately. The rule engine has to be restarted.

Reply:
Given that limitation, and as there are other bindings that do it this way, then I would argue to leave the package where it is (i.e. as a subpackage to `doorbird`). Ok?

Reply:
Yes, fine for me.